cmd/kube-rbac-proxy/app/kube-rbac-proxy.go

```diff
@@ -212,7 +212,15 @@ func Run(cfg *server.KubeRBACProxyConfig) error {
 		return fmt.Errorf("failed to setup an authorizer: %v", err)
 	}
 
-	proxy := httputil.NewSingleHostReverseProxy(cfg.KubeRBACProxyInfo.UpstreamURL)
+	proxy := &httputil.ReverseProxy{
+		Rewrite: func(pr *httputil.ProxyRequest) {
+			target := cfg.KubeRBACProxyInfo.UpstreamURL
+			pr.SetURL(target)
+			pr.Out.Host = target.Host
```

Comment:
Why do you set it, after rewrite removed it? To keep the same behavior as with the Director?

Reply:
Yes, I meant to keep the behavior the same.

Reply:
Yes, I assumed so, but I am wondering if this is some best practice like rewrite vs director :sweat_smile:

Reply:
Ok, I don't know why they are unsetting it, but it works with an empty request.Host, because of this: https://github.com/golang/go/blob/master/src/net/http/request.go#L605-L611. It will be automatically set, when the request is being written.

Reply:
I followed https://pkg.go.dev/net/http/httputil#NewSingleHostReverseProxy